internal/pkg/broker/broker_utils.go

```diff
@@ -0,0 +1,130 @@
+package broker
```

Comment:
Just calling this file `broker.go` is probably fine. Suffixing file names with `utils` makes it easy just to throw any function into it, and can quickly grow out of control (100+ lines of code)